errbot/botplugin.py

```diff
@@ -239,10 +244,15 @@ def stop_poller(self,
     def program_next_poll(self,
                           interval: float,
                           method: Callable[..., None],
+                          times: int=None,
                           args: Tuple=None,
                           kwargs: Mapping=None):
+        if type(times) in [int, float] and times <= 0:
```

Comment:
An assertion makes a lot of sense for such things. Though since we have type hinting, I'd argue even that is unnecessary now as `mypy` could be used to verify arguments of the correct type are provided.